file-formats/aifp/type/zif_aff_aifp_v1.intf.abap

```diff
@@ -0,0 +1,223 @@
+      message_variable_2 TYPE string,
+      "! <p class="shorttext">Message Variable 3</p>
+      "! Message variable 3
+      message_variable_3 TYPE string,
+      "! <p class="shorttext">Message Variable 4</p>
+      "! Message variable 4
+      message_variable_4 TYPE string,
+    END OF ty_message.
+  TYPES:
+    "! <p class="shorttext">Field Check</p>
+    "! Field check
+    BEGIN OF ty_field_check,
+      "! <p class="shorttext">Field Check</p>
+      "! Field check
+      field_check          TYPE ty_simple_field_check,
+      "! <p class="shorttext">Field Check Operator</p>
+      "! Field Check Operator
+      field_check_operator TYPE zif_aff_types_v1=>ty_option,
+      "! <p class="shorttext">Field Check Pattern</p>
+      "! Pattern for Field Check
+      field_check_pattern  TYPE c LENGTH 60,
+    END OF ty_field_check.
+  TYPES:
+    "! <p class="shorttext">Database Check</p>
+    "! Database check
+    BEGIN OF ty_database_check,
+      "! <p class="shorttext">Table Name</p>
+      "! Table name
+      table_name              TYPE c LENGTH 45,
+      "! <p class="shorttext">Where Condition for Select Statement</p>
+      "! Where Condition for Select Statement
+      where_condition         TYPE c LENGTH 200,
+      "! <p class="shorttext">Database Check Type</p>
+      "! Database check type
+      database_check_type     TYPE ty_database_check_type,
+      "! <p class="shorttext">Field Name for Comparison</p>
+      "! Field Name for Comparison
+      field_name              TYPE c LENGTH 45,
+      "! <p class="shorttext">Database Check Operator</p>
+      "! Database check operator
+      database_check_operator TYPE zif_aff_types_v1=>ty_option,
+      "! <p class="shorttext">Database Check Pattern</p>
+      "! Pattern for database check
+      database_check_pattern  TYPE c LENGTH 60,
+    END OF ty_database_check.
+  TYPES:
+    "! <p class="shorttext">Class Check</p>
+    "! Class check
+    BEGIN OF ty_class_check,
+      "! <p class="shorttext">Check Class</p>
+      "! Check class
+      check_class TYPE zif_aff_types_v1=>ty_object_name_30,
+    END OF ty_class_check.
+  TYPES:
+    "! <p class="shorttext">Single Check</p>
+    "! Single check
+    BEGIN OF ty_single_check,
+      "! <p class="shorttext">Check Number</p>
+      "! Check number
+      "! $required
+      check_number      TYPE n LENGTH 3,
+      "! <p class="shorttext">Check Description</p>
+      "! Check description
+      check_description TYPE c LENGTH 120,
+      "! <p class="shorttext">Check Scenario</p>
+      "! Check scenario
+      check_scenario    TYPE ty_check_scenario,
+      "! <p class="shorttext">Field Check</p>
+      "! Field check
+      field_check       TYPE ty_field_check,
+      "! <p class="shorttext">Database Check</p>
+      "! Database check
+      database_check    TYPE ty_database_check,
+      "! <p class="shorttext">Class Check</p>
+      "! Class check
+      class_check       TYPE ty_class_check,
```

Comment:
```suggestion
      class_check       TYPE zif_aff_types_v1=>ty_object_name_30,
```
is it better like this?
if you put the class check into an additional structure it will look different in the json and in the UI (editor)

Reply:
The division into substructures is intentional because, from an AIF perspective, there are three distinct types of checks: Field Checks, Database Checks, and Manual Implementations via Class. Depending on the type or combination selected in the scenario, the corresponding substructures should be displayed or hidden to facilitate maintenance.